server/src/KeycloakCrudService.ts

```diff
@@ -0,0 +1,102 @@
+import { CRUDService, GraphbackDataProvider, GraphbackPubSub } from "@graphback/runtime";
+import { GraphbackOperationType } from '@graphback/core';
+import { GraphQLObjectType } from 'graphql'
+
+export type CrudOperationAuthRules = {
+  roles: string[]
+}
+
+
+export type CrudServiceAuthRules = {
+  create: CrudOperationAuthRules,
+  read: CrudOperationAuthRules,
+  update: CrudOperationAuthRules,
+  delete: CrudOperationAuthRules,
+}
+
+export type KeycloakCrudServiceOptions = {
+  modelType: GraphQLObjectType,
+  db: GraphbackDataProvider,
+  subscriptionConfig: GraphbackPubSub,
+  authRules: CrudServiceAuthRules,
```

Comment:
I love how nice this fits to graphback - models.ts can contain representation of the auth rules